src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadExerciseResource.java

```diff
@@ -152,17 +150,17 @@ public FileUploadExerciseResource(FileUploadExerciseRepository fileUploadExercis
     }
 
     /**
-     * GET /file-upload-exercises/:id : get the "id" fileUploadExercise.
+     * GET /file-upload-exercises/:exerciseId : get the "id" fileUploadExercise.
      *
-     * @param id the id of the fileUploadExercise to retrieve
+     * @param exerciseId the id of the fileUploadExercise to retrieve
      * @return the ResponseEntity with status 200 (OK) and with body the fileUploadExercise, or with status 404 (Not Found)
      */
-    @GetMapping("/file-upload-exercises/{id}")
+    @GetMapping("/file-upload-exercises/{exerciseId}")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Transactional(readOnly = true)
-    public ResponseEntity<FileUploadExercise> getFileUploadExercise(@PathVariable Long id) {
-        log.debug("REST request to get FileUploadExercise : {}", id);
-        Optional<FileUploadExercise> fileUploadExercise = fileUploadExerciseRepository.findById(id);
+    public ResponseEntity<FileUploadExercise> getFileUploadExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get FileUploadExercise : {}", exerciseId);
+        Optional<FileUploadExercise> fileUploadExercise = fileUploadExerciseRepository.findById(exerciseId);
         Course course = fileUploadExercise.get().getCourse();
         User user = userService.getUserWithGroupsAndAuthorities();
         if (!authCheckService.isTeachingAssistantInCourse(course, user) && !authCheckService.isInstructorInCourse(course, user) && !authCheckService.isAdmin()) {
```

Comment:
You can just use the method `isAtLeastTeachingAssistantInCourse(course, user)` here instead. Also there is the method `isAtLeastInstructorInCourse(course, user)` which can be used for line 85 e.g.

Reply:
Good point, makes it easier to read code